nibabel/nifti1.py

```diff
@@ -528,6 +528,15 @@ def from_fileobj(klass, fileobj, size, byteswap):
             extensions.append(ext)
         return extensions
 
+# Temporary hack to avoid huge git diff
+Nifti1Extension.extension_codes = Nifti1Extensions.extension_codes
+
+class DeprecatedExtensionCodes(Nifti1Extensions):
+    @np.deprecate_with_doc("Use Nifti1Extensions.extension_codes")
+    def __getitem__(self, *args, **kwargs):
+        return super(DeprecatedExtensionCodes, self).__getitem__(*args, **kwargs)
+extension_codes = DeprecatedExtensionCodes.extension_codes
+
```

Comment:
Maybe [this model](https://github.com/nipy/nibabel/blob/master/nibabel/imageclasses.py#L90-L94) would be better? Instead of deprecating `__getitem__` on the recoder, you're deprecating it on a list that has a recoder as a class attribute.